src/Analyzers/MSTest.Analyzers/TestContextShouldBeValidAnalyzer.cs

```diff
@@ -92,11 +92,28 @@ private static bool AssignsParameterToMember(IParameterSymbol parameter, ISymbol
             operation = expressionStatementOperation.Operation;
         }
 
-        return operation is ISimpleAssignmentOperation assignmentOperation &&
+        if (operation is ISimpleAssignmentOperation assignmentOperation &&
             assignmentOperation.Target is IMemberReferenceOperation targetMemberReference &&
-            SymbolEqualityComparer.Default.Equals(targetMemberReference.Member, member) &&
-            assignmentOperation.Value is IParameterReferenceOperation parameterReference &&
-            SymbolEqualityComparer.Default.Equals(parameterReference.Parameter, parameter);
+            SymbolEqualityComparer.Default.Equals(targetMemberReference.Member, member))
+        {
+            // Handle direct parameter assignment
+            if (assignmentOperation.Value is IParameterReferenceOperation parameterReference &&
+                SymbolEqualityComparer.Default.Equals(parameterReference.Parameter, parameter))
+            {
+                return true;
+            }
+
+            // Handle null-coalescing operator with parameter on left side
+            // e.g., TestContext = testContext ?? throw new ArgumentNullException(nameof(testContext));
+            if (assignmentOperation.Value is ICoalesceOperation coalesceOperation &&
+                coalesceOperation.Value is IParameterReferenceOperation coalescingParamRef &&
+                SymbolEqualityComparer.Default.Equals(coalescingParamRef.Parameter, parameter))
+            {
+                return true;
+            }
```

Comment:
Instead of this, can you first "unwrap" the left side of `??`, then handle the `IParameterReferenceOperation` once?